### PR TITLE
chore: deprecate /superset/filter/... endpoint, migrate to apiv1

### PR DESCRIPTION
**Pull request by jfrag1**
### SUMMARY

Deprecates `/superset/filter/<datasource_type>/<datasource_id>/<column_name>/` endpoint

Creates a new, more REST-ful endpoint to replace it
* `/api/v1/datasource/<datasource_type>/<datasource_id>/column/<column_name>/values/`

Other improvements include better test coverage, better error handling, and an openapi spec

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/datasource/schemas.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from marshmallow import fields, Schema
+
+
+class GetColumnValuesResponseSchema(Schema):
+    result = fields.List(
+        fields.Raw(description="Can be any type, not just string"),
```
Comment (jfrag1) [inline superset/datasource/schemas.py:22]: spent ~15 minutes trying to figure out how to display an "any" type in swagger to no avail.  This shows up as string in the spec, so adding the description for clarification

**Comment by EugeneTorap** [comment]
Hi! Should we use `dataset` name instead of `datasource` in the new v1 API?

File: tests/integration_tests/datasource/api_tests.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+            assert val in response["result"]
+
+    @pytest.mark.usefixtures("app_context", "virtual_dataset")
+    def test_get_column_values_nulls(self):
+        self.login(username="admin")
+        table = self.get_virtual_dataset()
+        rv = self.client.get(f"api/v1/datasource/table/{table.id}/column/col4/values/")
+        self.assertEqual(rv.status_code, 200)
+        response = json.loads(rv.data.decode("utf-8"))
+        self.assertEqual(response["result"], [None])
```
Comment (dpgaspar) [inline tests/integration_tests/datasource/api_tests.py:73]: nit: all the above tests could be DRYer if we use `@pytest.mark.parametrize` (optional)

File: tests/integration_tests/datasource/api_tests.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        )
+        self.assertEqual(rv.status_code, 400)
+        response = json.loads(rv.data.decode("utf-8"))
+        self.assertEqual(response["message"], "Invalid datasource type: not_table")
+
+    @patch("superset.datasource.api.DatasourceDAO.get_datasource")
+    def test_get_column_values_datasource_type_not_supported(self, get_datasource_mock):
+        get_datasource_mock.side_effect = DatasourceTypeNotSupportedError
+        self.login(username="admin")
+        rv = self.client.get("api/v1/datasource/table/1/column/col1/values/")
```
Comment (dpgaspar) [inline tests/integration_tests/datasource/api_tests.py:90]: would sending `api/v1/datasource/bogus_type/1/column/col1/values/` result in the same?

**Review by dpgaspar — COMMENTED** [review]
LGTM, just a couple of details

**Comment by jfrag1** [comment]
> Hi! Should we use `dataset` name instead of `datasource` in the new v1 API?

Hi there! I think it makes sense to have a `/datasource` blueprint/namespace in the v1 API since datasets are a type of datasource, but there can be other types as well.

There are several "datasource APIs" in [this file](https://github.com/apache/superset/blob/master/superset/views/datasource/views.py) that I think can be moved here as well for 3.0.

It is somewhat confusing, but I think it would be more confusing to put all these under `/dataset`

File: superset/datasource/api.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+from superset.datasource.schemas import GetColumnValuesResponseSchema
+from superset.exceptions import SupersetSecurityException
+from superset.superset_typing import FlaskResponse
+from superset.utils.core import apply_max_row_limit, DatasourceType
+
+logger = logging.getLogger(__name__)
+
+
+class DatasourceRestApi(BaseApi):
+    include_route_methods = {"get_column_values"}
```
[the remark this answers is not in the document]
Comment (dpgaspar) [reply superset/datasource/api.py:34]: I'm finishing up this one: https://github.com/apache/superset/pull/22887 where I'm moving `BaseApi` to a new `BaseSupersetApi`, so now we can include statsd metrics to these endpoints

File: tests/integration_tests/core_tests.py [hunk 5]
```diff
@@ -325,21 +325,13 @@ def test_save_slice(self):
     @pytest.mark.usefixtures("load_energy_table_with_slice")
     def test_filter_endpoint(self):
         self.login(username="admin")
-        slice_name = "Energy Sankey"
-        slice_id = self.get_slice(slice_name, db.session).id
-        db.session.commit()
         tbl_id = self.table_ids.get("energy_usage")
         table = db.session.query(SqlaTable).filter(SqlaTable.id == tbl_id)
         table.filter_select_enabled = True
-        url = (
```
[the remark this answers is not in the document]
Comment (jfrag1) [reply tests/integration_tests/core_tests.py:334]: yeah, saw this and was like 🤨

File: tests/integration_tests/datasource/api_tests.py (hunk 3 above)
Comment (jfrag1) [reply tests/integration_tests/datasource/api_tests.py:90]: No, that would raise a `ValueError` on the endpoint, that case is covered by the `test_get_column_values_invalid_datasource_type` above this one.  This exception can be raised if the type is in the `DatasourceType` enum, but not supported by `get_datasource`

File: superset/datasource/schemas.py (hunk 1 above)
Comment (jfrag1) [reply superset/datasource/schemas.py:22]: removed and replaced with your suggestion

File: tests/integration_tests/datasource/api_tests.py (hunk 2 above)
Comment (jfrag1) [reply tests/integration_tests/datasource/api_tests.py:73]: Hadn't used that before, pretty neat.  However, I couldn't get it to play nice with it being a method on a class/using `self`.  I checked, and all the other tests in the codebase using that are standalone functions.  The duplication isn't too bad here, so opting to just leave as-is

**Review by dpgaspar — APPROVED** [review]

**Review by geido — APPROVED** [review]
Cypress change looks good

**Pull request merged**